### PR TITLE
chore: fix test cache inputs, clean up config

### PR DESCRIPTION
**Pull request by JamesHenry**
## Overview

Please see the notes inline for a description of the changes

### CODE REVIEW

File: .eslintrc.js
Comment (JamesHenry) [inline .eslintrc.js:1]: No longer needed

File: tsconfig.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
     "typings",
     "tools/**/*.ts",
     "tools/**/*.mts",
-    ".eslintrc.js",
```
Comment (JamesHenry) [inline tsconfig.json:12]: File removed

File: nx.json [hunk 2]
```diff
@@ -86,6 +80,12 @@
         "runtime": "yarn -v"
       }
     ],
-    "production": ["default", "!{projectRoot}/src/test-setup.[jt]s"]
+    "production": [
```
Comment (JamesHenry) [inline nx.json:83]: We were missing a bit more granularity on our definition of "production" files, we can exclude spec files and test specific config files from consideration

File: nx.json [hunk 3]
```diff
@@ -63,7 +58,6 @@
         "{workspaceRoot}/eslint.config.js",
         "{workspaceRoot}/eslint.config.mjs",
         "{workspaceRoot}/yarn.lock",
-        "{workspaceRoot}/.eslintignore",
```
Comment (JamesHenry) [inline nx.json:66]: File was already removed in a previous PR

File: nx.json [hunk 4]
```diff
@@ -40,6 +34,7 @@
     "@nx/jest:jest": {
       "inputs": [
         "default",
+        "^production",
```
Comment (JamesHenry) [inline nx.json:37]: This was important - we want the test tasks of projects to be affected when we make changes to the production files of dependencies (which is what the `^` denotes,  "of dependencies")

File: nx.json [hunk 5]
```diff
@@ -31,6 +24,7 @@
     "test": {
       "inputs": [
         "default",
+        "^production",
```
Comment (JamesHenry) [inline nx.json:27]: This was important - we want the test tasks of projects to be affected when we make changes to the production files of dependencies (which is what the `^` denotes,  "of dependencies")

File: nx.json [hunk 6]
```diff
@@ -16,10 +12,7 @@
       }
     },
     "version": {
-      "generatorOptions": {
```
Comment (JamesHenry) [inline nx.json:19]: Applying our newer shorthand for using conventional commits for versioning

**Review by bradzacher — APPROVED** [review]
You can remove the `eslint.config.js` now too and we can just have the `eslint.config.mjs`

**Comment by JamesHenry** [comment]
@bradzacher done

**Comment by JamesHenry** [comment]
@bradzacher scratch that, it looks like we need to handle `.mjs` flat config files on the nx side for this to work without the `eslint.config.js` being present, reverted for now so that we can land this. I'll update the Nx side

**Pull request merged**